src/main/kotlin/de/bigboot/ggtools/fang/service/MistforgeService.kt

```diff
@@ -73,11 +74,32 @@ class MistforgeService: AutostartService, KoinComponent {
         }
     }
 
-    private suspend fun handleMessage(event: MessageCreateEvent, buildId: String) {
-        val guide = mistforge.getGuide(buildId.toInt())
+    private suspend fun handleMessage(event: MessageCreateEvent, params: String) {
+        var guide: Guide? = null
+        val query_pairs = HashMap<String, String>()
+        if (params.contains("&")) {
+            val pairs = params.split("&")
+            for (pair in pairs) {
+                if (pair.contains("=")) {
+                    val idx = pair.indexOf("=")
+                    query_pairs.put(pair.substring(0, idx), pair.substring(idx + 1))
+                }
+            }
+        }
+        if (query_pairs.containsKey("build_id")) {
+            guide = mistforge.getGuide(query_pairs.get("build_id")!!.toInt())
+        }
 
         var img = background.copy()
 
+        var heroId = 0
```

Comment:
use `when` instead of this if/else thing, also rather return without parsing the url than setting some nonsense default